k8s/admissionregistration/admissionregistration.go

```diff
@@ -0,0 +1,137 @@
+package admissionregistration
+
+import (
+	"fmt"
+	"os"
+	"sync"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	apiadmissionsclient "k8s.io/client-go/kubernetes/typed/admissionregistration/v1beta1"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/clientcmd"
+)
+
+var (
+	instance Ops
+	once     sync.Once
+
+	deleteForegroundPolicy = metav1.DeletePropagationForeground
+)
+
+// Ops is an interface to perform kubernetes related operations admissionregistration resources.
+type Ops interface {
+	MutatingWebhookConfigurationOps
+
+	// SetConfig sets the config and resets the client.
+	SetConfig(config *rest.Config)
+}
+
+// Instance returns a singleton instance of the client.
+func Instance() Ops {
+	once.Do(func() {
+		if instance == nil {
```

Comment:
We don't need to do this nil check. `once.Do` ensure that the argument function is called only once in its lifetime.

Reply:
There is `SetInstance(i Ops)` func for each package that configures `instance`. So for such case `instance` value will be set to nil without the check:
```
i := fakeCoreClient{}
core.SetInstance(i)
core.Instance().ListPods() // here once.Do overrides the fake client
```